action/test/run-with-diff-id.test.ts

```diff
@@ -0,0 +1,222 @@
+import { run } from '../src/run';
+import { exec } from '@actions/exec';
+import { getInput, getMultilineInput, setFailed } from '@actions/core';
+import { octokit } from '../src/octokit';
+import { sync } from 'glob';
+import { BASE_IMAGES_DIRECTORY, NEW_IMAGES_DIRECTORY } from 'shared';
+import { expect } from '@jest/globals';
+
+jest.mock('../src/disableAutoMerge');
+jest.mock('glob');
+jest.mock('@actions/core');
+jest.mock('@actions/exec');
+jest.mock('@actions/github', () => ({
+  context: { repo: { repo: 'repo', owner: 'owner' } },
+  getOctokit: jest.fn(() => ({
+    rest: {
+      repos: {
+        createCommitStatus: jest.fn(),
+        listPullRequestsAssociatedWithCommit: jest.fn(),
+        listCommitStatusesForRef: jest.fn()
+      },
+      issues: {
+        createComment: jest.fn(),
+        listComments: jest.fn()
+      }
+    }
+  }))
+}));
+
+const inputMap: Record<string, string> = {
+  'screenshots-directory': 'path/to/screenshots',
+  'bucket-name': 'some-bucket',
+  'diff-id': 'uniqueId',
+  'github-token': 'some-token',
+  'comparadise-host': 'https://comparadise.app'
+};
+
+// Helper to assert no calls to `octokit.rest` methods
+const assertNoOctokitCalls = () => {
+  const allMethods = [
+    octokit.rest.repos.createCommitStatus,
+    octokit.rest.repos.listPullRequestsAssociatedWithCommit,
+    octokit.rest.repos.listCommitStatusesForRef,
+    octokit.rest.issues.createComment,
+    octokit.rest.issues.listComments
+  ];
+  allMethods.forEach(method => {
+    expect(method).not.toHaveBeenCalled();
+  });
+};
+
+describe('main with diff-id', () => {
```

Comment:
I added a separate test to keep the original tests simple. Combining this flow into the same file was making things a bit complicated. Not a big fan of duplication, but let me know your thoughts.

Reply:
Maybe now that the logic isn't duplicated it makes sense to keep these tests in the same file?